File: Rohit Singhal/LinkedLists/OddEvenLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        if(head== null || head.next==null || head.next.next==null)
        {
            return head;
        }
        
        
        ListNode oddstart=head;
        ListNode evenstart=head.next;
        ListNode odd=head;
        ListNode even=head.next;
        
        while(odd!=null && even!=null)
        {
            ListNode temp=even.next;
            if(even.next!=null)
            even.next=even.next.next;
            odd.next=temp;
            
            if(odd.next!=null)
            odd=odd.next;
            even=even.next;
        }
        odd.next=evenstart;
        return oddstart;
        
    }
}

File: Hashwanth/Linked Lists, Queues& Stacks/7_Odd Even Linked List/Odd Even Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
      if(!head || !head->next)
          return head;
     ListNode* o=head;
     ListNode* e=head->next;
     ListNode* eb=head->next;
     while(e && e->next)
     {
         o->next=o->next->next;
         e->next=e->next->next;
         o=o->next;
         e=e->next;
         
     }
        o->next=eb;
       return head;
    }
};
